SteamKit2/SteamKit2/Steam/CDN/Client.cs

```diff
@@ -0,0 +1,226 @@
+        {
+            if ( server == null )
+            {
+                throw new ArgumentNullException( nameof( server ) );
+            }
+
+            if ( chunk == null )
+            {
+                throw new ArgumentNullException( nameof( chunk ) );
+            }
+
+            if ( chunk.ChunkID == null )
+            {
+                throw new ArgumentException( "Chunk must have a ChunkID.", nameof( chunk ) );
+            }
+
+            var chunkID = Utils.EncodeHexString( chunk.ChunkID );
+
+            var chunkData = await DoRawCommandAsync( server, string.Format( "depot/{0}/chunk/{1}", depotId, chunkID ), proxyServer ).ConfigureAwait( false );
+
+            // assert that lengths match only if the chunk has a length assigned.
+            if ( chunk.CompressedLength > 0 && chunkData.Length != chunk.CompressedLength )
+            {
+                throw new InvalidDataException( $"Length mismatch after downloading depot chunk! (was {chunkData.Length}, but should be {chunk.CompressedLength})" );
+            }
+
+            var depotChunk = new DepotChunk( chunk, chunkData );
+
+            if ( depotKey != null )
+            {
+                // if we have the depot key, we can process the chunk immediately
+                depotChunk.Process( depotKey );
+            }
+
+            return depotChunk;
+        }
+
+        async Task<byte[]> DoRawCommandAsync( Server server, string command, Server? proxyServer )
+        {
+            var url = BuildCommand( server, command, proxyServer );
+            using var request = new HttpRequestMessage( HttpMethod.Get, url );
+
+            using var cts = new CancellationTokenSource();
+            cts.CancelAfter( RequestTimeout );
+
+            try
+            {
+                using var response = await httpClient.SendAsync( request, HttpCompletionOption.ResponseHeadersRead, cts.Token ).ConfigureAwait( false );
+
+                if ( !response.IsSuccessStatusCode )
+                {
+                    throw new SteamKitWebRequestException( $"Response status code does not indicate success: {response.StatusCode:D} ({response.ReasonPhrase}).", response );
+                }
+
+                // .NET 5.0 has an override of ReadAsByteArrayAsync that allows a CancellationTokenSource to be supplied
+                cts.CancelAfter( ResponseBodyTimeout );
+
+                var contentLength = response.Content.Headers.ContentLength;
+
+                using var responseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait( false );
+                using var ms = new MemoryStream( ( int )contentLength.GetValueOrDefault() );
+
+                await responseStream.CopyToAsync( ms, 81920, cts.Token ).ConfigureAwait( false );
+
+                return ms.ToArray();
+            }
+            catch ( Exception ex )
+            {
+                DebugLog.WriteLine( "CDNClient", "Failed to complete web request to {0}: {1}", url, ex.Message );
```

Comment:
`"CDNClient"` isn't a class anymore, so we're now diverging from a category == a class.